Mathlib/LinearAlgebra/Dimension/StrongRankCondition.lean

```diff
@@ -542,3 +542,19 @@ theorem mem_span_set_iff_exists_finsupp_le_finrank :
     exact mem_span_set.mpr ⟨c, hcs, hx⟩
 
 end Submodule
+
+section IsQuadraticAlgebra
+
+/--
+An extension of rings `R ⊆ S` is quadratic if `S` is a free `R`-module of rank `2`.
+-/
+-- TODO. use this in connection with `NumberTheory.Zsqrtd`
+class IsQuadraticAlgebra (R S : Type*) [CommSemiring R] [Semiring S] [Algebra R S]
+  extends StrongRankCondition R, Module.Free R S where
```

Comment:
I think we can put `StrongRankCondition R` as a parameter, so in practice people will just forget about it.

Reply:
Just to make sure I understand what you mean: you want me to remove `StrongRankCondition R` as a prerequisite for `IsQuadraticAlgebra` and just add it as an extra hypothesis in the results that need it ?

Reply:
I mean putting `[StrongRankCondition R]` next to `[Algebra R S]`. In practice it is always satisfied (by any `CommRing`).

Reply:
Oh, sorry. I understand now.